topostats/plottingfuncs.py

```diff
@@ -14,108 +14,212 @@
+        Returns
+        -------
+        fig: plt.figure.Figure
+            Matplotlib.pyplot figure object
+        ax: plt.axes._subplots.AxesSubplot
+            Matplotlib.pyplot axes object
+        """
+        fig, ax = None, None
+        if self.save:
+            if self.image_set == "all" or self.core_set:
+                if self.axes or self.colorbar:
+                    fig, ax = self.save_figure()
+                else:
+                    if isinstance(self.data2,np.ndarray) or self.region_properties:
+                        fig, ax = self.save_figure()
+                    else:
+                        self.save_array_figure()
+        if "_processed" in self.filename:
+            LOGGER.info(f"[{self.filename.split('_processed')[0]}] : Image saved to : {str(self.output_dir / self.filename)}")
+        return fig, ax
+
+    def save_figure(self):
+        """
+        This function saves figures as plt.savefig objects.
+        
+        Returns
+        -------
+        fig: plt.figure.Figure
+            Matplotlib.pyplot figure object
+        ax: plt.axes._subplots.AxesSubplot
+            Matplotlib.pyplot axes object
+        """
+        fig, ax = plt.subplots(1, 1, figsize=(8, 8))
+        shape = self.data.shape
+        if isinstance(self.data, np.ndarray):
+            im = ax.imshow(
+                self.data,
+                extent=(0, shape[0] * self.pixel_to_nm_scaling_factor, 0, shape[1] * self.pixel_to_nm_scaling_factor),
+                interpolation=self.interpolation,
+                cmap=Colormap(self.cmap).get_cmap(),
+                vmin=self.zrange[0],
+                vmax=self.zrange[1],
             )
-            patch = [Patch(color=plt.get_cmap("jet_r")(1, 0.7), label="Mask")]
-            plt.legend(handles=patch, loc="upper right", bbox_to_anchor=(1, 1.06))
+            if isinstance(self.data2, np.ndarray):
+                mask = np.ma.masked_where(self.data2 == 0, self.data2)
+                ax.imshow(
+                    mask,
+                    "jet_r",
+                    extent=(0, shape[0] * self.pixel_to_nm_scaling_factor, 0, shape[1] * self.pixel_to_nm_scaling_factor),
+                    interpolation=self.interpolation,
+                    alpha=0.7,
+                )
+                patch = [Patch(color=plt.get_cmap("jet_r")(1, 0.7), label="Mask")]
+                plt.legend(handles=patch, loc="upper right", bbox_to_anchor=(1, 1.06))
 
-        plt.title(title)
-        plt.xlabel("Nanometres")
-        plt.ylabel("Nanometres")
-        if colorbar and type == "non-binary":
-            divider = make_axes_locatable(ax)
-            cax = divider.append_axes("right", size="5%", pad=0.05)
-            plt.colorbar(im, cax=cax, label="Height (Nanometres)")
-        if region_properties:
-            fig, ax = add_bounding_boxes_to_plot(fig, ax, region_properties, pixel_to_nm_scaling_factor)
+            plt.title(self.title)
+            plt.xlabel("Nanometres")
```

Comment:
With an eye on the future SI units the `xlabel` and `ylabel` values should probably be configurable options, whether that's when you `__init__` the class or for the `save_figure` method itself.

Reply:
I've been thinking about this too and think it would be useful to use the dictionary made in `Filters.extract_pixel_to_nm_scaling()` to carry units with the scaling objects to then place into plotting.

But again, I think this should be left for a future PR to not mix up things. I believe it's only been pulled up here due to the restructuring.